Restore Internet-NOTAMs via FAA JSON API

b-spatz opened the pull request:
Restore Internet-provided NOTAMs for location via map longpress using FAA API per https://notams.aim.faa.gov/notamSearch/.

Includes LTA (Letter to Airmen) as FAA page does (useful but often overlooked by pilots but much appreciated by ATC).
Could exclude LTAs (line 31) or simplify (by removing text URL, line 32), or make URL an active link per methods you prefer (RichText, etc).
Example LTAs for GNV area:
JAX approach: https://notams.aim.faa.gov/lta/main/viewlta?lookupid=3657797174768440930
GNV tower: https://notams.aim.faa.gov/lta/main/viewlta?lookupid=3626657331632149547

Restoring NOTAM capability is very important as I want to easily check a few NOTAMs before every flight and the AvareX map-based interface is much quicker and direct than FAA page above.

Happy Thanksgiving!

apps4av commented:
- Some testing is needed at checking Notams at KBVY and KPVD hangs the Notam screen with spinner. 
- The LTA links are not clickable. Thats OK but they should at least be cut-paste enabled. Try copy text field.

b-spatz commented:
Updated notam_cache.dart to strengthen JSON parsing.  Simplified LTA display per original PR; defer to your call on presentation functionality: make URL clickable for copy/paste or launch.  For now, just making more pilots aware is a Good Thing.

b-spatz commented:
On review of this PR, realized I forgot to include the LTA simplification...added above.  Sorry for any confusion.  Do I need to resubmit the entire PR?
